pcdsdevices/dccm.py

```diff
@@ -0,0 +1,289 @@
+import logging
+import typing
+from collections import namedtuple
+from typing import Optional
+
+import numpy as np
+from ophyd.device import Component as Cpt
+from ophyd.device import FormattedComponent as FCpt
+
+from .beam_stats import BeamEnergyRequest
+from .device import GroupDevice
+from .epics_motor import BeckhoffAxis
+from .interface import BaseInterface, FltMvInterface
+from .pseudopos import PseudoPositioner, PseudoSingleInterface
+
+logger = logging.getLogger(__name__)
+
+# Constants
+si_111_dspacing = 3.1356011499587773
+
+# Defaults
+default_dspacing = si_111_dspacing
+
+
+class DCCMEnergy(FltMvInterface, PseudoPositioner):
+    """
+    DCCM energy motor.
+
+    Calculates the current DCCM energy using the DCCM angle, and
+    requests moves to the DCCM motors based on energy requests.
+
+    Presents itself like a motor.
+
+    Parameters
+    ----------
+    prefix : str
+        The PV prefix of the DCCM motor, e.g. XPP:MON:MPZ:07A
+    """
+    # Pseudo motor and real motor
+    energy = Cpt(
+        PseudoSingleInterface,
+        egu='keV',
+        kind='hinted',
+        limits=(4, 25),
+        verbose_name='DCCM Photon Energy',
+        doc=(
+            'PseudoSingle that moves the calculated DCCM '
+            'selected energy in keV.'
+        ),
+    )
+
+    th1 = Cpt(BeckhoffAxis, ":MMS:TH1", doc="Bragg Upstream/TH1 Axis", kind="normal", name='th1')
+    th2 = Cpt(BeckhoffAxis, ":MMS:TH2", doc="Bragg Upstream/TH2 Axis", kind="normal", name='th2')
+
+    tab_component_names = True
+
+    def forward(self, pseudo_pos: namedtuple) -> namedtuple:
```

Comment:
Nit: we could possibly be more specific with the namedtuple hint here, if we're bothering to hint it at all